### ISSUE TITLE
MapLibre instead of MapBox

### ISSUE THREAD

**Issue by diyoyo** [photoview/photoview#879]
**Is your feature request related to a problem? Please describe.**
I'm starting using photoview because I want a self-hosted solution that is free and open-source. Using MapBox doesn't seem to comply with my constraints.

**Describe the solution you'd like**
Seems like MapLibre is a viable alternative, from what I understand.

**Describe alternatives you've considered**
Not sure, I took the first result from StackExchange to propose a solution.

**Additional context**
https://github.com/maplibre/maplibre-gl-js

**Comment by apple314159** [comment]
Mapbox provides a library and a tile service. MapLibre is a library and it does not provide tiles. So this suggestion is not really viable.

**Comment by dasKeks28** [comment]
I also would love an alternative tiling provider instead of MapBox which requires to add a credit card even for the free tier.
I'm afraid that some extensive usage / bug would just push me out of the free tier which becomes expensive quickly.
I'd rather have a limit and upon reaching tiles cannot be loaded instead of becoming non-free.

I'd suggest MapTiler which is also used by Photoprism.

**Comment by hjonin** [comment]
I can also suggest https://www.thunderforest.com/.

**Comment by kkovaletp** [comment]
There is a discussion #709 about the same topic. We need to check it as well at the solution design time

**Comment by louwers** [comment]
> MapLibre is a library and it does not provide tiles. So this suggestion is not really viable.

Correct, MapLibre is a library, but there are many tile providers available.

Some free of charge like https://openfreemap.org/ and https://protomaps.com/ (free for non-commercial use). Commercial offerings are available from for example MapTiler, JawgMaps and Radar. See some options here: https://madewithmaplibre.com/basemaps/gallery

Full disclosure: I am a maintainer for MapLibre.

Let me know if you have any questions!
